superset/daos/dashboard.py

```diff
@@ -63,8 +63,6 @@ def get_by_id_or_slug(cls, id_or_slug: int | str) -> Dashboard:
             query = (
                 db.session.query(Dashboard)
                 .filter(id_or_slug_filter(id_or_slug))
-                .outerjoin(Slice, Dashboard.slices)
```

Comment:
@dpgaspar would you mind looking into this as it seems you added this logic in https://github.com/apache/superset/pull/22358? Note it's interesting that the unit/integration tests passed with the removal of these two lines.